contracts/interchain-token-service/src/msg.rs

```diff
@@ -60,8 +59,8 @@ pub struct ChainConfig {
 
 #[cw_serde]
 pub struct TruncationConfig {
-    pub max_uint: nonempty::Uint256, // The maximum uint value that is supported by the chain's token standard
-    pub max_decimals_when_truncating: u8, // The maximum number of decimals that is preserved when deploying from a chain with a larger max_uint
+    pub max_uint_bits: u32, // The maximum number of bits used by the chain to represent unsigned integers
```

Comment:
Do we have a `nonempty::u32`? zero wouldn't make sense here.

Reply:
if we are going to change the type, I'd rather add more restrictions than just non zero. While zero is wrong, I think many other u32 values are also wrong, such as 1 or other small values, odd values probably, values that are not a power of 2, and values greater than 256.

Reply:
i created a new type for this, lmk what you think